### DISCUSSION TITLE
In Directory Sheet, is there a shortcut for opening a recursive listing for a subfolder in the sheet itself?

### DISCUSSION THREAD

**Discussion by reagle** [saulpw/visidata#1847] [Q&A] [answered]
I see we can recurse on command line argument; is it possible to do so when opening a subfolder sheet in the app?

Also, could I use this for cleaning file names? For example, replacing spaces with underscores and lowercasing file names?

**Comment by saulpw** [comment, marked as answer]
There's no way to set `options.dir_recurse` in the UI, but we could easily make a command like gEnter or something on the DirSheet that pushes the copy and sets that option on it before loading.  Should be a one-liner.

And yes, you can use it for cleaning file names.  Make sure to set the `filename` column directly; just making a new column named `filename` isn't sufficient.  You'll see the colors change to indicate there are uncommitted (deferred) changes.  Use `z Ctrl+S` to commit them once they look right.

**Reply by reagle** [reply]
Ah yes, `gz=filename.lower().replace(' ', '_')` works nicely. I think `gEnter` sounds good.

**Reply by saulpw** [reply]
(I think you mean `g=` instead of `gz=`)

**Reply by reagle** [reply]
Yes, thank you. (I haven't figured out `gz=` yet.)
